command/certificate/remote.go

```diff
@@ -56,20 +63,27 @@ func getPeerCertificates(addr, serverName, roots string, insecure bool) ([]*x509
 // trimURL returns the host[:port] if the input is a URL, otherwise returns an
 // empty string (and 'isURL:false').
 //
+// If the URL is valid and no port is specified, the default port determined
+// by the URL prefix is used.
+//
 // Examples:
-// trimURL("https://smallstep.com/onbaording") -> "smallstep.com", true, nil
+// trimURL("https://smallstep.com/onbaording") -> "smallstep.com:443", true, nil
 // trimURL("https://ca.smallSTEP.com:8080") -> "ca.smallSTEP.com:8080", true, nil
 // trimURL("./certs/root_ca.crt") -> "", false, nil
-// trimURL("hTtPs://sMaLlStEp.cOm") -> "sMaLlStEp.cOm", true, nil
+// trimURL("hTtPs://sMaLlStEp.cOm") -> "sMaLlStEp.cOm:443", true, nil
 // trimURL("hTtPs://sMaLlStEp.cOm hello") -> "", false, err{"invalid url"}
 func trimURL(ref string) (string, bool, error) {
 	tmp := strings.ToLower(ref)
-	for _, prefix := range urlPrefixes {
+	for prefix, _ := range urlPrefixes {
 		if strings.HasPrefix(tmp, prefix) {
 			u, err := url.Parse(ref)
 			if err != nil {
 				return "", false, errors.Wrapf(err, "error parsing URL '%s'", ref)
 			}
+			if _, _, err := net.SplitHostPort(u.Host); err != nil {
```

Comment:
For the sake of consistency, please use and return the `err` object, even if it would logically be handled by the previous one.

Reply:
I'm not sure I understand.

I think this error should not be returned, it is the same code as :

https://github.com/smallstep/cli/blob/26bd65c9670b30eb5407f1e311bafe2858a27592/command/certificate/remote.go#L45-L47

Please clarify what you want :smiley:

Reply:
I mean do a similar `err` return as the [lines above](https://github.com/smallstep/cli/blob/26bd65c9670b30eb5407f1e311bafe2858a27592/command/certificate/remote.go#L40-L43)

Reply:
This looks correct as is since the err is used to detect the case where there is no port provided.